src/mlpack/methods/random_forest/bootstrap.hpp

```diff
@@ -43,6 +45,153 @@ void Bootstrap(const MatType& dataset,
+           typename WeightsType>
+  void Bootstrap(const MatType& dataset,
+                 const LabelsType& labels,
+                 const WeightsType& weights,
+                 MatType& bootstrapDataset,
+                 LabelsType& bootstrapLabels,
+                 WeightsType& bootstrapWeights)
+  {
+    mlpack::Bootstrap<UseWeights>(dataset,
+                                  labels,
+                                  weights,
+                                  bootstrapDataset,
+                                  bootstrapLabels,
+                                  bootstrapWeights);
+  }
+};
+
+/**
+ * No-op bootstrap strategy that just copies the input to the output.
+ */
+template<bool UseWeights>
+class IdentityBootstrap
+{
+ public:
+  template<typename MatType,
+           typename LabelsType,
+           typename WeightsType>
+  void Bootstrap(const MatType& dataset,
+                 const LabelsType& labels,
+                 const WeightsType& weights,
+                 MatType& bootstrapDataset,
+                 LabelsType& bootstrapLabels,
+                 WeightsType& bootstrapWeights)
+  {
+    bootstrapDataset = dataset;
+    bootstrapLabels = labels;
+    if (UseWeights)
+      bootstrapWeights = weights;
+  }
+};
+
+/**
+ * Sequential bootstrap.
+ */
+template<bool UseWeights, typename G = std::mt19937>
+class SequentialBootstrap
+{
+ public:
+  explicit SequentialBootstrap(const arma::mat& indicatorMatrix,
+                               const G& generator = G(std::random_device()())) :
+    SequentialBootstrap(indicatorMatrix, indicatorMatrix.n_rows, generator)
+  {
+  }
+
+  explicit SequentialBootstrap(const arma::mat& indicatorMatrix,
+                               arma::uword sampleCount,
+                               const G& generator = G(std::random_device()())):
+    indicatorMatrix(indicatorMatrix),
+    colIndices(arma::linspace<arma::uvec>(0, indicatorMatrix.n_cols - 1)),
+    sampleCount(sampleCount),
+    generator(generator)
+  {
+  }
+
+  template<typename MatType,
+           typename LabelsType,
+           typename WeightsType>
+  void Bootstrap(const MatType& dataset,
+                 const LabelsType& labels,
+                 const WeightsType& weights,
+                 MatType& bootstrapDataset,
+                 LabelsType& bootstrapLabels,
+                 WeightsType& bootstrapWeights)
+  {
+    if (labels.n_cols != indicatorMatrix.n_rows)
+      throw std::invalid_argument("SequentialBootstrap::Bootstrap(): "
+          "labels and indicatorMatrix n_rows differ!");
+
+    // observations are stored as columns and dimensions
+    // (number of features) as rows.
+
+    const arma::uvec phi(ComputeSamples());
+
+    bootstrapDataset = dataset.cols(phi);
+    bootstrapLabels = labels.cols(phi);
+    if (UseWeights)
+      bootstrapWeights = weights.rows(phi);
+  }
+
+  arma::uvec ComputeSamples() const
+  {
+    arma::uvec phi;
+
+    while (phi.size() < sampleCount) {
```

Comment:
```suggestion
    while (phi.size() < sampleCount)
    {
```
[style fix](https://github.com/mlpack/mlpack/wiki/DesignGuidelines#style-guidelines)